code/game/turfs/simulated.dm

```diff
@@ -18,26 +19,64 @@
 
 	roof_type = /turf/simulated/floor/airless/ceiling
 
-/turf/simulated/proc/wet_floor(var/wet_val = 1)
-	if(wet_val < wet)
-		return
-
-	if(!wet)
-		wet = wet_val
+/turf/simulated/proc/wet_floor(var/type = WET_TYPE_WATER, var/amount = 1)
+
+	//Wet type:
+	//WET_TYPE_WATER = water
+	//WET_TYPE_LUBE = lube
+	//WET_TYPE_ICE = ice
+	
+	if(!wet_type)
+		wet_type = type
+	else if(type != wet_type)
+		if(type == WET_TYPE_WATER) //Water is being added

```

Comment:
This could be a switch, and I suspect there's a more concise mathematical way to do this.